### DISCUSSION TITLE
SG Ready

### DISCUSSION THREAD

**Discussion by Chair-for-Processor-Design-TU-Dresden** [fredlcore/BSB-LAN#583] [General]
I have been using BSB-Lan to monitor my heat pump (Brötje Sensotherm BSW K 8B) through Openhab binding for a good 4 years.

I would like to use my excess solar power to trigger the heat pump. Does someone have any idea whether it is possible via BSB-LAN interface? My heat pump is supposedly SG-ready (Smartgrid Ready) and should be able to start heating or circulation, whenever the SG Ready signal is received. However, I have no idea who or how one should send such a signal.

Does anyone know whether any parameter can be set for this? Can one build a (simple) hardware module/relay for this? 

Thanks a lot in advance!

**Comment by lemon75** [comment]
Hello,
just implementing such a feature here for my elco heat pump.
In the documentation the input is described: for me it is a 230 V input which needs to be set to 230 V to tell the heat pump the PV inverter has surplus energy.
In my case I already wired a solid state relay which has a 5 V input for control. This input will be set/reset by the ESP32 which runs the BSB-LAN SW. But I still need to code the SW part to allow my smart-home solution to use this properly.
Hope that helps.
Regards,
Jan

**Comment by fredlcore** [comment]
I'm sorry that I didn't get back to this discussion earlier.
As for this matter, I'd first look at the different ways to control the heat pump's activity. For example, would it be sufficient to increase the minimum set temperature for buffer (assuming that the heat pump charges a buffer). If that is the case, I'd increase the buffer's minimum set temperature by 10 degrees or so if there is excess PV energy available.
The second thing I'd check is if one of the potential-free inputs (such as H1 or H3) give you the option to define this input to force the heatpump to charge the buffer. The beauty of this is that you would not even need an actual relay to open/close the input, but you could just define the input as active open or active closed with another parameter. For example, I use H1 to put my heater into standby mode (parameter 5950). I used to do this with a relay (controlled by BSB-LAN), but now I just change parameter 5951 ("Wirksinn") to change the "Wirksinn" to inverted and back again, depending on which way I want H1 to work.

If there is really only a 230V input available (which would surprise me), then you could use the Olimex's built-in relays that allow for 220V, if I'm not mistaken. You can control them (on the Olimex) with URL-commands /G32=1 or /G32=0 to turn them on and off (/G33 works likewise).

**Comment by jdreesma** [comment]
I am working currently on the same topic. As I have a BSW 10 E with the same Controler ISR RVS 61.843-100 as in the BSW K8B.
But I only checked this by reading manuals on the Broetje website. So please double check it ;-)
Suggesting this is your installation manual:
https://polo.broetje.de/pdf/c0027579ge_ih_bsw-k_7300939-06.pdf
please have a look on page 163 -165. Section "Konfiguration" Topic "Funktion Eingang EX1-4 (5980-5992)".
So I suggesting Input ports EX1 and EX2 coud be configured to the SG1 and SG2, what is by factory default, "EW Sperre E6" and "Niedertarif E5" AND important hint... you have to define if the ports are working as NC (normaly close) or NO (normally open) Contact! EW Sperre is by default NO!!
!! please refer your Schematic Diagram in the topcover of your Heatpump!!
Also you have to set 230V on these Input ports , so you should be a electrician to connect the to avoid electrical shock...
The SG Ready Signal should be send by a wide range of photovoltaic inverters or smart managers who observe the 230V power flow and if you have a overflow and could not use it by yourselfe it probably will sent it in the Power grid.
Then the SG Reagy signal could start the Heatpump to store the Power in Warm Water or in the House itselve. 1 or 2 degrees

**Comment by fredlcore** [comment]
Thanks! I think this is the way I thought it would/should work, except that I was hoping/thinking that inverters would/could send a potential-free signal which could then be used as an input for H1 etc. without the need for an electrician. But if a 230V signal is the norm for inverters (should check mine as well at some point), then the way you are describing is the most suitable one.

**Comment by fredlcore** [comment]
It's been a while, but since another user has been working on a similar direction, here is another idea:
As you said, defining the EX1/EX2 to block or force the heat pump to heat the buffer („Electric supplier lock E6“ and „Low tariff E5“, respectively) in parameters 5980 and 5982 is the key here. But the good thing is that you don’t actually need to hard wire the SG-ready signal from the inverter to those inputs. You can also control their behaviour via parameters 5981 and 5983: By switching between „normal close“ and „normal open“, you also switch between the actual functions. If you don’t need blocking, only forcing, you can also just do it with one of these inputs, for example setting 5982 to „Low tariff E5“ and then activate it by setting 5983 to „open“ and deactivate it by setting it to „close“.

This way, you can also programmatically check if the inverter has excess PV energy, and then change parameter 5983 accordingly. Thus, you don't need a wired solution, and it even works with those inverters that don't provide an SG-ready signal, such as mine.

I haven't tested it myself, but since it is the same way that I use to control functions via the H1 dry contact, it should work. But if anyone has really tested it, please report back here!

**Reply by kekeseb** [reply]
I have just implemented this together with EVCC and it's working perfectly

**Reply by fredlcore** [reply]
What exactly did you do? Do you mind sharing your workflow here, as I'm sure others will appreciate it!

**Reply by kekeseb** [reply]
I've only tested with DHW at the moment, but I need to experiment a bit more with heating to figure out what makes sense.

Basically, I have `1620 DHW - DHW release`set to `4 - Time switch program 4 or Low-tariff`

This is my heat pump config in EVCC.
- The home assistant switch toggles parameter `5987 Configuration - Operating action input EX4` using MQTT
- I monitor the power usage with a Shelly Pro 3 EM meter: https://docs.evcc.io/en/docs/devices/meters#shelly-1

```## required attributes [type: sgready-boost]

charger: # configuration of a switch controlling the sg-ready port
  type: template
  template: homeassistant-switch
  baseurl: http://homeassistant.local/ # Base URL of the Home Assistant instance
  token: 
  switchentity: switch.heat_pump_sg_ready_boost # Entity ID of the switch device

## optional attributes (read-only)

temp: # current temperature (°C)
  source: http
  uri: "http://bsb-lan.local/JQ=8830"
  jq: ."8830".value
limittemp: # temperature limit (°C) configured in device
  source: http
  uri: "http://bsb-lan.local/JQ=8831"
  jq: ."8831".value
```

**Reply by fredlcore** [reply]
Thanks!
